src/node_env_var.cc

```diff
@@ -389,13 +390,31 @@ static void EnvEnumerator(const PropertyCallbackInfo<Array>& info) {
       env->env_vars()->Enumerate(env->isolate()));
 }
 
+static void EnvDefiner(Local<Name> property,
+                       const PropertyDescriptor& desc,
+                       const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  if (desc.has_get() || desc.has_set()) {
+    env->ThrowError("Cannot use getter or setter on 'process.env'");
+  } else if (desc.has_value()) {
```

Comment:
You should also throw an error when `!(desc.configurable() && desc.enumerable() && desc.writable())` because that's behavior we can't/won't emulate.